atlas/probe_cache.go

```diff
@@ -17,13 +17,9 @@ func InitCache(ttl, cleanup time.Duration) {
 
 func startCacheCleanupFunc(d time.Duration) {
 	go func() {
-		for {
```

Comment:
By removing the for loop the cache would only be cleaned once after start. Did you want to simplify the loop body and returned the loop iteself instead? 

Rest looks good.

Reply:
Dropped the change for getting this merged. :-)